src/row_codec.rs

```diff
@@ -1,182 +1,230 @@
-use crate::db_state::RowFeature;
 use crate::error::SlateDBError;
-use crate::types::{KeyValueDeletable, RowAttributes, ValueDeletable};
+use crate::types::{RowEntry, ValueDeletable};
 use bitflags::bitflags;
 use bytes::{Buf, BufMut, Bytes, BytesMut};
 
 bitflags! {
+    #[derive(Debug, Clone, PartialEq, Default)]
     pub(crate) struct RowFlags: u8 {
         const Tombstone = 0b00000001;
+        const HAS_EXPIRE_TS = 0b00000010;
+        const HAS_CREATE_TS = 0b00000100;
     }
 }
 
-const NO_EXPIRE_TS: i64 = i64::MIN;
-
 /// Encodes key and value using the binary codec for SlateDB row representation
-/// using the `v0` encoding scheme.
+/// using the `v1` encoding scheme.
 ///
-/// The `v0` codec for the key is (for non-tombstones):
+/// The `v1` codec for the key is (for non-tombstones):
 ///
 /// ```txt
-///  |--------------------------------------------------------------------------|
-///  |       u16      |      u16       |  var        | var  |    u32    |  var  |
-///  |----------------|----------------|-------------|------|-----------|-------|
-///  | key_prefix_len | key_suffix_len |  key_suffix | meta | value_len | value |
-///  |--------------------------------------------------------------------------|
+///  |---------------------------------------------------------------------------------------------------------------------|
+///  |       u16      |      u16       |  var        | u64     | u8        | i64       | i64       | u32       |  var      |
+///  |----------------|----------------|-------------|---------|-----------|-----------|-----------|-----------|-----------|
+///  | key_prefix_len | key_suffix_len |  key_suffix | seq     | flags     | expire_ts | create_ts | value_len | value     |
+///  |---------------------------------------------------------------------------------------------------------------------|
 /// ```
 ///
-/// And for tombstones (with `meta[row_flags] & 0x01 == 1`):
+/// And for tombstones (flags & Tombstone == 1):
+///
 ///  ```txt
-///  |------------------------------------------------------|
-///  |       u16      |      u16       |  var        | var  |
-///  |----------------|----------------|-------------|------|
-///  | key_prefix_len | key_suffix_len |  key_suffix | meta |
-///  |------------------------------------------------------|
+///  |----------------------------------------------------------|-----------|
+///  |       u16      |      u16       |  var        | u64      | u8        |
+///  |----------------|----------------|-------------|----------|-----------|
+///  | key_prefix_len | key_suffix_len |  key_suffix | seq      | flags     |
```

Comment:
in line with the comment above, it's still possible for tombstones to have `create_ts` associated with them (there's some good reasons, especially for "as of" timestamp queries), so let's just keep `meta` here as well